test/integration/types_test.rb

```diff
@@ -103,28 +106,33 @@ def test_json_data_insert
     # with a JSON column is broken on 5.7.
     # See: https://bugs.mysql.com/bug.php?id=87847
     res = target_db.query("SELECT COUNT(*) AS cnt FROM #{DEFAULT_FULL_TABLE_NAME}")
-    assert_equal 16, res.first["cnt"]
+    assert_equal ENV["MYSQL_VERSION"] == "8.0" ? 22 : 18, res.first["cnt"] # 22 rows with trailing zero data, 18 without trailing zero
```

Comment:
Is there any other better way? checking and counting manually sounds not like a sustainable/maintainable solution here

Reply:
https://github.com/Shopify/ghostferry/pull/368/commits/98d4b922d81770a521f84816898fa91e7fb5d101 - changes here I think its better?